src/query/stream/stream_consumer_factory.cpp

```diff
@@ -0,0 +1,237 @@
+  auto result = mgp_result{nullptr, memory_resource};
+
+  return [result = std::move(result), memory_resource, interpreter_context, interpreter = std::move(interpreter),
+          stream_name, transformation_name,
+          owner = std::move(owner)](const std::vector<typename TStream::Message> &messages) mutable {
+#ifdef MG_ENTERPRISE
+    interpreter->OnChangeCB([](auto) { return false; });  // Disable database change
+#endif
+    auto accessor = interpreter->current_db_.db_acc_->get()->Access();
+    // register new interpreter into interpreter_context
+    interpreter_context->interpreters->insert(interpreter.get());
+    memgraph::utils::OnScopeExit interpreter_cleanup{
+        [interpreter_context, interpreter]() { interpreter_context->interpreters->erase(interpreter.get()); }};
+
+    memgraph::metrics::IncrementCounter(memgraph::metrics::MessagesConsumed, messages.size());
+
+    CallCustomTransformation(transformation_name, messages, result, *accessor, *memory_resource, stream_name);
+
+    memgraph::query::DiscardValueResultStream stream;
+
+    spdlog::trace("Start transaction in stream '{}'", stream_name);
+    memgraph::utils::OnScopeExit cleanup{[&interpreter, &result]() {
+      result.rows.clear();
+      interpreter->Abort();
+    }};
+
+    uint32_t i = 0;
+    auto const total_retries = interpreter_context->config.stream_transaction_conflict_retries;
+    const auto retry_interval = interpreter_context->config.stream_transaction_retry_interval;
+    while (true) {
+      try {
+        interpreter->BeginTransaction();
+        for (auto &row : result.rows) {
+          spdlog::trace("Processing row in stream '{}'", stream_name);
+          auto [query_value, params_value] = ExtractTransformationResult(row.values, transformation_name, stream_name);
+          memgraph::storage::PropertyValue params_prop{params_value};
+
+          std::string query{query_value.ValueString()};
+          spdlog::trace("Executing query '{}' in stream '{}'", query, stream_name);
+          auto prepare_result = interpreter->Prepare(
+              query,
+              params_prop.IsNull() ? std::map<std::string, memgraph::storage::PropertyValue>{} : params_prop.ValueMap(),
+              {});
+          if (!interpreter_context->auth_checker->IsUserAuthorized(owner, prepare_result.privileges, "")) {
+            throw StreamsException{
+                "Couldn't execute query '{}' for stream '{}' because the owner is not authorized to execute the "
+                "query!",
+                query, stream_name};
+          }
+          interpreter->PullAll(&stream);
+        }
+
+        spdlog::trace("Commit transaction in stream '{}'", stream_name);
+        interpreter->CommitTransaction();
+        result.rows.clear();
+        break;
+      } catch (const memgraph::query::TransactionSerializationException &e) {
+        interpreter->Abort();
+
+        if (i == total_retries) {
```

Comment:
You can use here ResettableCounter you added last week right?

Reply:
For this usage, it was moved code from elsewhere. I do not ATM want to use ResettableCounter just in case I have the retry count wrong and change behaviour to be off by one.

For now I'll leave as is.